llama.cpp

```diff
@@ -3110,9 +3110,17 @@ struct llama_model_loader {
 
         fver = (enum llama_fver) gguf_get_version(meta);
 
+        std::set<std::string> tensor_names;
         for (auto & w : weights) {
             n_elements += ggml_nelements(w.tensor);
             n_bytes    += ggml_nbytes(w.tensor);
+            // make sure there is no duplicated tensor names
+            const std::string name(w.tensor->name);
+            auto found = tensor_names.find(name);
+            if (found != tensor_names.end()) {
+                LLAMA_LOG_ERROR("%s: Found duplicated tensor name %s", __func__, w.tensor->name);
```

Comment:
It's not enough to print a message, it also needs to throw.

```suggestion
                throw std::runtime_error(format("invalid model: tensor '%s' is duplicated", w.tensor->name));
```